src/service/url-replacements-impl.js

```diff
@@ -1077,11 +1077,15 @@ export class UrlReplacements {
     if (element[ORIGINAL_HREF_PROPERTY] == null) {
       element[ORIGINAL_HREF_PROPERTY] = href;
     }
+
+    const isAllowedOrigin = this.isAllowedOrigin_(url);
```

Comment:
Let's return early if the origin is not allowed.

Reply:
Sounds good

Reply:
Oops, looks like we can't resolve early. [Some of the tests](https://github.com/ampproject/amphtml/blob/master/test/unit/test-url-replacements.js#L1902-L1943) indicate that even if it's not an allowed origin, the `data-amp-addparams` should still be appended.

Reply:
👍 Good call. LGTM